### ISSUE TITLE
tf2.4.1 : acc is different from end epoch and modelcheckpoint acc 

### ISSUE THREAD

**Issue by zhaoyun0071** [keras-team/keras#14514]
_**my code**_ 
`import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Activation, Dropout
from tensorflow.keras.utils import to_categorical, plot_model
from tensorflow.keras.datasets import mnist
from tensorflow.keras.callbacks import EarlyStopping, ReduceLROnPlateau, ModelCheckpoint

model_name = '{epoch:02d}-{accuracy:.4f}-{val_accuracy:.4f}--{loss:.4f}.h5'

(x_train, y_train), (x_test, y_test) = mnist.load_data()

num_labels = len(np.unique(y_train))

y_train = to_categorical(y_train)
y_test = to_categorical(y_test)

image_size = x_train.shape[1]
input_size = image_size * image_size

x_train = np.reshape(x_train, [-1, input_size])
x_train = x_train.astype('float32') / 255
x_test = np.reshape(x_test, [-1, input_size])
x_test = x_test.astype('float32') / 255

batch_size = 128
hidden_units = 256
dropout = 0.45

model = Sequential()
model.add(Dense(hidden_units, input_dim=input_size))
model.add(Activation('relu'))
model.add(Dropout(dropout))
model.add(Dense(hidden_units))
model.add(Activation('relu'))
model.add(Dropout(dropout))
model.add(Dense(num_labels))

model.add(Activation('softmax'))
model.summary()

model.compile(loss='categorical_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=20, batch_size=batch_size, callbacks=[
        ModelCheckpoint(
            model_name,
            monitor='val_accuracy',
            verbose=1,
            mode='max',
            save_best_only=True)],validation_split=0.5)
`

and epoch 1 below:

`235/235 [==============================] - 4s 12ms/step - loss: 0.982742 - accuracy: 0.677682 - val_loss: 0.216704 - val_accuracy: 0.933267

Epoch 00001: val_accuracy improved from -inf to 0.93327, saving model to 01-0.8166-0.9333--0.5841.h5
Epoch 2/20`

_**i can see loss: 0.982742 - accuracy: 0.677682 on 1 epoch end

but modelcheckpoint is acc:0.8166  loss :0.5841

why????

this code in tf 1.15.2 have no problems

what changes in tf2?**_

**Comment by jvishnuvardhan** [comment]
@zhaoyun0071 I couldn't reproduce the issue with `TF2.5`. May be there was some issue with older version that was corrected in the recent stable version. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/c46022fa212ed2ac6a65b7a3c716220a/untitled1007.ipynb).

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing this issue as this was resolved. Please feel free to reopen if I am mistaken. Thanks!
